### PR TITLE
Adding the chi square distribution

### PR DESCRIPTION
**Pull request by talgalili**
Adding the chi square distribution, which is central to many statistical procedures (https://en.wikipedia.org/wiki/Chi-square_distribution) (#15798)

Test plan (adding unit-tests)

Following the diff template of: https://github.com/prestodb/presto/pull/11981/files

```
== RELEASE NOTES ==

General Changes
* Add chisquared_cdf() and inverse_chisquared_cdf() functions.

(like the beta_cds: https://prestodb.io/docs/current/release/release-0.215.html?highlight=beta_cds)

### CODE REVIEW

**Comment by talgalili** [comment]
Thanks, good catch. I'll fix and resubmit later today (/or early tomorrow).
Any other comments to address?

T

File: presto-docs/src/main/sphinx/functions/math.rst [hunk 1]
```diff
@@ -85,6 +85,18 @@ Mathematical Functions
     The a, b parameters must be positive real numbers and value v must be a real value.
     The value v must lie on the interval [0, 1].
 
+.. function:: inverse_chisquared_cdf(df, p) -> double
+
+    Compute the inverse of the Chi-square cdf with given df (degrees of freedom) parameter for the cumulative
+    probability (p): P(N < n). The the df parameter must be positive real values.
```
[the remark this answers is not in the document]
Comment (talgalili) [reply presto-docs/src/main/sphinx/functions/math.rst]: @v-jizhang  - fixed. Thanks.
Please let me know if you have any more suggetions.

**Comment by talgalili** [comment]
Hey @v-jizhang ,
tl;dr: I think this diff is ready for review / merging.

Details:
I've fixed the type in the doc, as well as found/fixed a bug in the testing. As far as I can see it all seems to work fine now. There is still some test that didn't finish running (but it doesn't seem related to this diff).
I'll re-base the diff, and have the tests re-run.

Cheers,
T

**Comment by talgalili** [comment]
(rebased)

**Review by v-jizhang — APPROVED** [review]
Looks good to me.

File: presto-docs/src/main/sphinx/functions/math.rst [hunk 2]
```diff
@@ -85,6 +85,18 @@ Mathematical Functions
     The a, b parameters must be positive real numbers and value v must be a real value.
     The value v must lie on the interval [0, 1].
 
+.. function:: inverse_chi_squared_cdf(df, p) -> double
```
Comment (rongrong) [inline presto-docs/src/main/sphinx/functions/math.rst:88]: Please put the function documentation in alphabetical order. The cdf documentation were in wrong places. You don't need to fix them in this PR, just make sure the new function documentations are added at the correct place.

File: presto-docs/src/main/sphinx/functions/math.rst [hunk 3]
```diff
@@ -166,7 +178,7 @@ Mathematical Functions
 .. function:: truncate(x, n) -> double
 
     Returns ``x`` truncated to ``n`` decimal places.
-    ``n`` can be negative to truncate ``n`` digits left of the decimal point. 
+    ``n`` can be negative to truncate ``n`` digits left of the decimal point.
```
Comment (rongrong) [inline presto-docs/src/main/sphinx/functions/math.rst:181]: Remove.

File: presto-main/src/main/java/com/facebook/presto/operator/scalar/MathFunctions.java [hunk 4]
```diff
@@ -704,6 +705,32 @@ public static double betaCdf(
...
 
+    @Description("inverse of ChiSquared cdf given df parameter and probability")
+    @ScalarFunction
+    @SqlType(StandardTypes.DOUBLE)
+    public static double inverseChiSquaredCdf(
+            @SqlType(StandardTypes.DOUBLE) double df,
+            @SqlType(StandardTypes.DOUBLE) double p)
+    {
+        checkCondition(p >= 0 && p <= 1, INVALID_FUNCTION_ARGUMENT, "p must be in the interval [0, 1]");
+        checkCondition(df > 0, INVALID_FUNCTION_ARGUMENT, "df must be > 0");
```
Comment (rongrong) [inline presto-main/src/main/java/com/facebook/presto/operator/scalar/MathFunctions.java:716]: `must be greater than 0`

**Review by rongrong — COMMENTED** [review]
Looks good. Please fix the nits. Thanks for contributing!

**Comment by talgalili** [comment]
Thanks Rongrong!
I'll fix it early next week and will ping once it's done.

T

File: presto-main/src/main/java/com/facebook/presto/operator/scalar/MathFunctions.java (hunk 4 above)
Comment (talgalili) [reply presto-main/src/main/java/com/facebook/presto/operator/scalar/MathFunctions.java:716]: Done.

File: presto-docs/src/main/sphinx/functions/math.rst (hunk 2 above)
Comment (talgalili) [reply presto-docs/src/main/sphinx/functions/math.rst:88]: Given that we intend to add more distributions (t, F, Gamma, etc.). I think a better approach is to have a dedicated section in the doc called "Probability distributions" (similar to how the doc currently has "Statistical Functions", "Trigonometric Functions", etc.) 

I can do it in a separate diff after we finish adding all the distribution functions. Makes sense?

File: presto-docs/src/main/sphinx/functions/math.rst (hunk 3 above)
Comment (talgalili) [reply presto-docs/src/main/sphinx/functions/math.rst:181]: Needed to turn "trimTrailingWhitespace" off in the VS code setting.
Done (although this trailing white space should probably be removed regardless).

**Comment by talgalili** [comment]
@rongrong - thanks for the diff review.
I've addressed all the issues you've raised (except for the location of the help doc, please see my suggestion above).

Assuming it's all good, please move forward with merging this diff.
Once I see you gave this the green light, me and others from the hackathon will proceed to making all other "distribution function" go through the same workflow.

Thanks!

**Comment by talgalili** [comment]
Hey @rongrong :)
Any update on reviewing this diff? Can it be merged?

File: presto-docs/src/main/sphinx/functions/math.rst (hunk 2 above)
Comment (rongrong) [reply presto-docs/src/main/sphinx/functions/math.rst:88]: Makes sense. You can add a separate PR to refactor the documentation first, and rebase this PR on top of that one. Or if you prefer to do that afterwards, let's put these functions in the conventional order for this PR and modify it later.

**Comment by talgalili** [comment]
Great.
I'll modify this diff (in the coming two hours) to follow your proposed
order, and will submit a new diff to create a new section in the doc in tge
coming weeks.

Will update soon.

T

**Comment by talgalili** [comment]
Hey @rongrong - I've now:
1. re-positioned the function documentation in the (hopefully) correct location
2. re-based and updated the PR

Please let me know if this now works.

Once Merged, I'll update the other contributors of all the learnings gained in this PR for me, and we'll proceed with preparing more diffs.

Yours,
T

**Review by rongrong — APPROVED** [review]
